fw/connection.h

```diff
@@ -518,6 +524,26 @@ tfw_connection_validate_cleanup(TfwConn *conn)
 	BUG_ON(rc && rc != TFW_CONN_DEATHCNT);
 }
 
+static inline int
+tfw_peer_for_each_conn(TfwPeer *p, int (*cb)(TfwConn *))
+{
+	int r = 0;
+	TfwConn *conn, *tmp_conn;
+
+	spin_lock_bh(&p->conn_lock);
+
+	/* @cb() may delete connections from the list. */
+	list_for_each_entry_safe(conn, tmp_conn, &p->conn_list, list) {
+		r = cb(conn);
```

Comment:
I guess there should `list_del()` invocation somewhere inside the `cb(conn)`, but I don't see any, e.g in `__cli_conn_close_cb()` / `__cli_conn_abort_cb()` / `tfw_sock_srv_disconnect()`

Reply:
I added better comment for the function: theclient connection drop callback removes the entry from the list.